Changes for cassandra opentelemetry doc after with metric collection … (#4309)
SumoLogic/sumologic-documentation: docs/integrations/databases/opentelemetry/cassandra-opentelemetry.md

<img src={useBaseUrl('img/integrations/databases/cassandra.png')} alt="Thumbnail icon" width="60"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

> The [Cassandra](https://cassandra.apache.org/_/cassandra-basics.html) - OpenTelemetry app is a log and metric based app that helps you monitor the availability, performance, health, and resource utilization of your Cassandra clusters. Preconfigured dashboards provide insight into resource utilization, cache/Gossip/Memtable statistics and Error and warnings along with the request served and latency, storage and compaction related metrics. Cassandra logs are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver) and cassandra metrics are sent to Sumo Logic using [JMX opentelemetry receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/jmxreceiver) with the `target_system` set as [`cassandra`](https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/jmx-metrics/docs/target-systems/cassandra.md)

Reviewer: ```suggestion
The [Cassandra](https://cassandra.apache.org/_/cassandra-basics.html) app is a log and metrics-based app that helps you monitor the availability, performance, health, and resource utilization of your Cassandra clusters. The pre-configured dashboards provide insight into resource utilization, cache/Gossip/Memtable statistics, Error and Warnings , along with the request served and latency, storage and compaction related metrics. 

Cassandra logs are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver) and cassandra metrics are sent to Sumo Logic using [JMX opentelemetry receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/jmxreceiver) with the `target_system` set as [`cassandra`](https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/jmx-metrics/docs/target-systems/cassandra.md).
```

- `deployment.environment`. User configured. Through this Cassandra cluster is identified by the environment where it resides. For example: dev, prod or qa.
- `db.cluster.name`. User configured. Enter a name to identify this Cassandra cluster. This cluster name will be shown in the Sumo Logic dashboards.
- `db.system`. Has fixed value of **cassandra**.
- `db.node.name`. Holds the value of the Fully Qualified Domain Name (FQDN) of the machine from which the OpenTelemetry collector is collecting logs and metrics.
- `sumo.datasource`. Has fixed value of **cassandra**.

## Prerequisites

#### For metric collection

JMX receiver collects Cassandra metrics from Cassandra server as part of the OpenTelemetry Collector (OTC).

  1. Follow the instructions in [JMX - OpenTelemetry's prerequisites section](/docs/integrations/app-development/opentelemetry/jmx-opentelemetry/#prerequisites) to download the [JMX Metric Gatherer](https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/jmx-metrics/README.md). This gatherer is used by the [JMX Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/jmxreceiver#details).

>   2. Set the JMX port by setting it as part of `JAVA_OPTS` for Tomcat startup. Usually it is set in the `/etc/systemd/system/cassandra.service` or `C:\Program Files\apache-tomcat\bin\tomcat.bat` file.

Reviewer: ```suggestion
  2. Set the JMX port as part of `JAVA_OPTS` for Tomcat startup. Usually, it is set in the `/etc/systemd/system/cassandra.service` or `C:\Program Files\apache-tomcat\bin\tomcat.bat` file.
```

  INFO [ScheduledTasks:1] 2023-01-08 09:18:47,347 StatusLogger.java:101 - system.schema_aggregates
```

> ## Sample metric message

Reviewer: ```suggestion
## Sample metrics message
```

```json
{
    "queryId": "A",
    "_source": "cassandra-otel",
    "_metricId": "dasIFS6RvCVxEcsljWGpnQ",
    "_sourceName": "Http Input",
    "host.id": "i-071ade75d03650657",
    "host.group": "cassandraHostGrp",
    "telemetry.sdk.version": "1.38.0",
    "os.type": "linux",
    "sumo.datasource": "cassandra",
    "db.system": "cassandra",
    "db.node.name": "ip-54-163-607-30.ec2.internal",
    "_sourceCategory": "Labs/cassandra-otel",
    "deployment.environment": "dev",
    "_contentType": "Carbon2",
    "host.name": "ip-54-163-607-30.ec2.internal",
    "metric": "cassandra.client.request.write.latency.50p",
    "_collectorId": "000000000FB3B451",
    "_sourceId": "000000005A60F624",
    "unit": "us",
    "db.cluster.name": "cassandra_cluster",
    "telemetry.sdk.language": "java",
    "telemetry.sdk.name": "opentelemetry",
    "service.name": "unknown_service:java",
    "_collector": "Labs - cassandra-otel",
    "max": 229,
    "min": 0,
    "avg": 131.25,
    "sum": 525,
    "latest": 67,
    "count": 4
}
```

> ## Sample Log queries 

Reviewer: ```suggestion
## Sample log queries 
```

| sort by _timeslice asc
```

> ## Sample Metrics queries 

Reviewer: ```suggestion
## Sample metrics query
```

Following is the query from Cassandra App's overview Dashboard's Number of Requests Panel:

```sql
sumo.datasource=cassandra deployment.environment=* db.cluster.name=* db.node.name=* metric=cassandra.client.request.count | sum 
```

> ## Viewing Cassandra Dashboards

Reviewer: ```suggestion
## Viewing Cassandra dashboards
```

- Identify resource utilization. This can help you to determine whether resources are over-allocated or under-allocated.

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Cassandra-OpenTelemetry/Cassandra-Resource-Usage-Logs.jpg' alt="Resource Usage" />

### Compaction

> The Cassandra - Compactions dashboard provides a insight into the completed and pending compaction tasks.

Reviewer: ```suggestion
The **Cassandra - Compactions** dashboard provides insight into the completed and pending compaction tasks.
```

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Cassandra-OpenTelemetry/Cassandra-Compaction.jpg' alt="Compaction" />

### Requests

> The Cassandra - Requests dashboard gives insight into the number of request served, number of error request and their distribution by status and operation. Also you can monitor the read and write latency of the cluster instance using this dashboard.

Reviewer: ```suggestion

The **Cassandra - Requests** dashboard provides insight into the number of request served, number of error request, and their distribution by status and operation. Also you can monitor the read and write latency of the cluster instance using this dashboard.
```

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Cassandra-OpenTelemetry/Cassandra-Requests.jpg' alt="Requests" />

### Storage

> The Cassandra - Storage dashboard provides insight into the current value of total hints of your cassandra cluster along with storage managed by the cluster.

Reviewer: ```suggestion

The **Cassandra - Storage** dashboard provides insight into the current value of total hints of your Cassandra cluster along with storage managed by the cluster.
```
